### PR TITLE
Set ignore done=False in GAIL

### PR DESCRIPTION
**Pull request by andrewcoh**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [x] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [x] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [x] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [x] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Review by awjuliani — APPROVED** [review]
LGTM!

File: config/imitation/PushBlock.yaml [hunk 1]
```diff
@@ -16,16 +16,24 @@ behaviors:
       num_layers: 2
       vis_encode_type: simple
     reward_signals:
-      gail:
+      extrinsic:
         gamma: 0.99
         strength: 1.0
+      gail:
+        gamma: 0.99
```
Comment (vincentpierre) [inline config/imitation/PushBlock.yaml:23]: Should this be lower ?

File: ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py [hunk 2]
```diff
@@ -21,12 +21,13 @@
...
     def __init__(self, specs: BehaviorSpec, settings: GAILSettings) -> None:
         super().__init__(specs, settings)
-        self._ignore_done = True
+        self._ignore_done = False
         self._discriminator_network = DiscriminatorNetwork(specs, settings)
         self._discriminator_network.to(default_device())
         _, self._demo_buffer = demo_to_buffer(
             settings.demo_path, 1, specs
         )  # This is supposed to be the sequence length but we do not have access here
+        self._discriminator_network.encoder.update_normalization(self._demo_buffer)
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py:30]: How long will this take for large demo sets? Can this become prohibitive and if yes, should we use a sample of the demo buffer ?

**Review by vincentpierre — APPROVED** [review]

File: ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py (hunk 2 above)
Comment (andrewcoh) [reply ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py]: I replaced this with samples from the expert demos in the update

File: config/imitation/PushBlock.yaml (hunk 1 above)
Comment (andrewcoh) [reply config/imitation/PushBlock.yaml:23]: I don't believe so. My feeling is that we decrease strength not gamma to curb the effect of GAIL

File: docs/ML-Agents-Overview.md [hunk 3]
```diff
@@ -504,6 +515,7 @@ actions. In addition to learning purely from demonstrations, the GAIL reward
 signal can be mixed with an extrinsic reward signal to guide the learning
 process.
 
+
```
Comment (ervteng) [inline docs/ML-Agents-Overview.md:518]: Should this line be here?

File: ml-agents/mlagents/trainers/settings.py [hunk 4]
```diff
@@ -183,6 +183,7 @@ def to_settings(self) -> type:
 class RewardSignalSettings:
     gamma: float = 0.99
     strength: float = 1.0
+    normalize: bool = False
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/settings.py:186]: We should put this in GAILRewardSignalSettings for now, until we implement it in Curiosity as well.

Comment (andrewcoh) [reply ml-agents/mlagents/trainers/settings.py:186]: resolved in https://github.com/Unity-Technologies/ml-agents/pull/4982

**Review by ervteng — APPROVED** [review]
Seems fine once the NetworkSettings are merged

**Pull request merged**